src/internal/predicates.mjs

```diff
@@ -29,3 +31,27 @@ export function isIterator(i){
 export function isArray(x){
   return Array.isArray(x);
 }
+
+export function hasMethod(method, x){
+  return Boolean(x) && typeof x[method] === 'function';
```

Comment:
Why `Boolean(x)` rather than `x != null`? It seems wrong to me:

```javascript
> (function (method, x) { return Boolean (x) && typeof x[method] === 'function'; } ('concat', 'xyz'))
true

> (function (method, x) { return Boolean (x) && typeof x[method] === 'function'; } ('concat', ''))
false
```

Reply:
That is a good point. I've been using `Boolean` to check if something is "accessible", but I hadn't considered strings, numbers and Booleans.

Reply:
`null` and `undefined` are the only “inaccessible” values, as far as I'm aware.